processor/src/main/java/com/linecorp/decaton/processor/runtime/ProcessorSubscription.java

```diff
@@ -263,6 +266,13 @@ private void consumeLoop() {
             contexts.updateHighWatermarks();
             try {
                 commitManager.commitSync();
+            } catch (RebalanceInProgressException e) {
+                // At this point, Decaton already stopped processing though, from KafkaConsumer perspective,
+                // it's not closed yet (i.e. rebalance due to member-leave is not initiated), so another rebalance might be
+                // initiated (likely by other consumer instances), and then, this commitSync may fail due to
+                // RebalanceInProgressException.
+                // So, since it's not a fatal situation and kind of expected failure, we just log at WARN level.
```

Comment:
> "failing to commit at the end of termination" itself is an unexpected situation for users

Maybe?
However, I found that with cooperative rebalancing, "shutdown (offset commit)-during-rebalance" happens easily than eager. (I can see RebalanceInProgressException every time I run `testCooperativeRebalancing` integration test)

I'll check further and decide what would be the best here.

Reply:
> I can see RebalanceInProgressException every time I run testCooperativeRebalancing integration test

Figured out the cause.
Integration test does:
1. start 3 subscriptions one by one, with awaiting each subscription transitions to `RUNNING`
2. Start rolling restart of subscriptions. This includes stopping a subscription and then start it.

The problem is, with cooperative rebalancing, "state transitioned to `RUNNING`" is no longer a sign of rebalance stabilized.
With cooperative rebalancing, typical rebalance by new-member addition causes 2 rounds of consumer rebalance.
One for revoking the partition from existing member, one for assign them to newly added member. This is to ensure single partition never be assigned to multiple instances.

So, if step 2 above (rolling-restart phase) starts before finishing second round of member-addition-rebalance, when rolling restarting, Decaton may fail to commit-sync on shutdown due to `RebalanceInProgressException`. That's what happened in `testCooperativeRebalancing`.

Reply:
So, unless this (https://issues.apache.org/jira/browse/KAFKA-9382) is addressed, users may see more offset commit failure on application rolling restart due to this in cooperative rebalancing.

Anyways, I agree with keeping it `ERROR` because it means duplicate-processing will likely occur, which users would not expect for usual rolling restart.